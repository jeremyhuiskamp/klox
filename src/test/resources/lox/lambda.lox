fun thrice(fn) {
  for (var i = 1; i <= 3; i = i + 1) {
    fn(i);
  }
}

thrice(fun (a) {
  print a;
});

// this should be ignored:
fun () {};

// but this is a bit more fun:
print fun(a) { return a; }("hello lambda");
fun(a) { print a; }("lambda hello");

// works nicely with unary operators, call syntax
// binds stronger:
print ! fun(a) { return a; }(false);

// yikes, we don't even need braces around the body ðŸ˜¬
print fun(a) return !a; (true);
