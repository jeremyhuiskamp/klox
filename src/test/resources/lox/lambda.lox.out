1
2
3
hello lambda
lambda hello
true
false